nexus/db-queries/src/db/queries/vpc_subnet.rs

```diff
@@ -549,4 +715,140 @@ mod test {
+    #[tokio::test]
+    async fn test_filter_conflicting_vpc_subnet_ranges_query_is_idempotent() {
+        let ipv4_block = "172.30.0.0/24".parse().unwrap();
+        let other_ipv4_block: oxnet::Ipv4Net = "172.30.1.0/24".parse().unwrap();
+        let ipv6_block = "fd12:3456:7890::/64".parse().unwrap();
+        let other_ipv6_block: oxnet::Ipv6Net =
+            "fd12:3456:7800::/64".parse().unwrap();
+        let name = "a-name".to_string().try_into().unwrap();
+        let description = "some description".to_string();
+        let identity = IdentityMetadataCreateParams { name, description };
+        let vpc_id = "d402369d-c9ec-c5ad-9138-9fbee732d53e".parse().unwrap();
+        let subnet_id = "093ad2db-769b-e3c2-bc1c-b46e84ce5532".parse().unwrap();
+        let row =
+            VpcSubnet::new(subnet_id, vpc_id, identity, ipv4_block, ipv6_block);
+
+        // Setup the test database
+        let logctx = dev::test_setup_log(
+            "test_filter_conflicting_vpc_subnet_ranges_is_idempotent",
+        );
+        let log = logctx.log.new(o!());
+        let mut db = test_setup_database(&log).await;
+        let cfg = crate::db::Config { url: db.pg_config().clone() };
+        let pool = Arc::new(crate::db::Pool::new(&logctx.log, &cfg));
+        let db_datastore = Arc::new(
+            crate::db::DataStore::new(&log, Arc::clone(&pool), None)
+                .await
+                .unwrap(),
+        );
+
+        // We should be able to insert anything into an empty table.
+        let inserted = db_datastore
+            .vpc_create_subnet_raw(row.clone())
+            .await
+            .expect("Should be able to insert VPC subnet into empty table");
+
+        // Helper to verify equality of rows, handling timestamp precision.
+        let assert_rows_eq = |left: &VpcSubnet, right: &VpcSubnet| {
+            assert_eq!(
+                left.identity.id, right.identity.id,
+                "VPC Subnet rows should be equal"
+            );
+            assert_eq!(
+                left.identity.name, right.identity.name,
+                "VPC Subnet rows should be equal"
+            );
+            assert_eq!(
+                left.identity.description, right.identity.description,
+                "VPC Subnet rows should be equal"
+            );
+            // Timestamps in CRDB have microsecond precision, so ensure we're
+            // within 1000 nanos.
+            assert!(
+                (left.identity.time_modified - right.identity.time_modified)
+                    .num_nanoseconds()
+                    .unwrap()
+                    < 1_000,
+                "VPC Subnet rows should be equal",
+            );
+            assert!(
+                (left.identity.time_created - right.identity.time_created)
+                    .num_nanoseconds()
+                    .unwrap()
+                    < 1_000,
+                "VPC Subnet rows should be equal",
+            );
```

Comment:
oof, looks like they aren't even truncating, but *rounding* to the nearest microsecond. I think what you have here is fine, but historically I have ignored a lot of "timestamp comparisons" in tests for this reason. That said, I suppose this is a more extensive check!

Reply:
Yeah, I think ignoring them is OK in many cases too. I kinda wonder if we should provide some methods for comparisons that ignore or check the difference between two timestamps, say as part of deriving the `Identity` trait or similar. But not critical :)